TLM/TLM/Manager/Impl/LaneArrowManager.cs

```diff
@@ -143,30 +143,36 @@ public class LaneArrowManager
         /// Narrow down updated roads to those that can have dedicated turning lanes.
         /// </param>
         public void UpdateAllDefaults(bool dedicatedTurningLanes) {
+            Log.Info("UpdateAllDefaults(dedicatedTurningLanes:{dedicatedTurningLanes}) was called.");
             SimulationManager.instance.AddAction(delegate () {
-                for (ushort segmentId = 1; segmentId < NetManager.MAX_SEGMENT_COUNT; ++segmentId) {
-                    ref NetSegment segment = ref segmentId.ToSegment();
-                    if (!netService.IsSegmentValid(segmentId))
-                        continue;
-
-                    if (segment.Info?.GetAI() is not RoadBaseAI ai)
-                        continue;
-
-                    int forward = 0, backward = 0;
-                    segmentId.ToSegment().CountLanes(segmentId, LANE_TYPES, VEHICLE_TYPES, ref forward, ref backward);
-                    if (dedicatedTurningLanes && forward == 1 && backward == 1) {
-                        // one lane cannot have dedicated turning lanes.
-                        continue;
-                    }
-
-                    if (dedicatedTurningLanes &&
-                        segment.m_startNode.ToNode().CountSegments() <= 2 &&
-                        segment.m_endNode.ToNode().CountSegments() <= 2) {
-                        // no intersection.
-                        continue;
+                try {
+                    Log._Debug($"Executing UpdateAllDefaults(dedicatedTurningLanes:{dedicatedTurningLanes}) in simulation thread ...");
+                    for (ushort segmentId = 1; segmentId < NetManager.MAX_SEGMENT_COUNT; ++segmentId) {
+                        ref NetSegment segment = ref segmentId.ToSegment();
+                        if (!netService.IsSegmentValid(segmentId))
+                            continue;
+
+                        if (segment.Info?.GetAI() is not RoadBaseAI ai)
+                            continue;
+
+                        int forward = 0, backward = 0;
+                        segmentId.ToSegment().CountLanes(segmentId, LANE_TYPES, VEHICLE_TYPES, ref forward, ref backward);
+                        if (dedicatedTurningLanes && forward == 1 && backward == 1) {
+                            // one lane cannot have dedicated turning lanes.
+                            continue;
+                        }
+
+                        if (dedicatedTurningLanes &&
```

Comment:
`if (dedicatedTurningLanes)` is checked inside the loop for every iteration, have it before the loop.

Reply:
Yes, two usages of that method currently and both pass `true` as argument, but I think I see a bug (You should pass `value` instead of `true` - `OptionsVehicleRestrictionsTab.cs`)
![image](https://user-images.githubusercontent.com/19638970/127561095-d435bd5c-2265-4926-a659-cf5d90045910.png)

Reply:
> @krzychu124  You should pass value instead of true

that value shows if the toggle is true or false and is only there to match Delegate parameters.
whether the option is true or false we only need to update dedicated turning lanes.

I am going to get rid of the parameter and change the names accordingly.

Reply:
Wait.. I'm not sure we are on the same page. `LaneArrowsUpdateHandler` is attached to checkbox click... 
In the current code there is no difference if I switch `enabled -> disabled` or `disabled -> enabled` because you always call `UpdateAllDefaults(true)` after change -> dedicated turning lanes. 
The question is if it was meant to work like that?

Reply:
yes.
both disabled->enabled and enabled->disabled need to call `UpdateAllDefaults(true)`.

Reply:
OK, so to force update, right? If so, like @kvakvs said, that param seem to be redundant as you always pass 'true' as argument to that method.